(CDAP-16353) Replace ArtifactRepository param with PluginFinder for AbstractConfigurer…

wyzhang opened the pull request:
(CDAP-16353) Replace ArtifactRepository param with PluginFinder for AbstractConfigurer

 Motivation:
AbstractConfigurer uses ArtifactRepository just to find the artifact for a given plugin,
this functionality is already provided by PluginFinder which has local and remote implementations,
thereby allowing us to use local impl when having shared SQL and remote version for non-shared
per service local levelDB.

wyzhang commented:
https://builds.cask.co/browse/CDAP-DUT7159-5

All tests passed.

wyzhang commented:
Note this PR doesn't introduce any change to functionality. For now, all pluginFinders are bound to LocalPluginFinder, which is the same as using ArtifactRepository for accessing local store prior to this change.

yaojiefeng commented on cdap-app-fabric/src/main/java/io/cdap/cdap/app/guice/AppFabricServiceRuntimeModule.java:
```diff
@@ -21,6 +21,7 @@
 import com.google.common.collect.ImmutableList;
 import com.google.inject.AbstractModule;
 import com.google.inject.Module;
+import com.google.inject.PrivateModule;
```
Unused?

wyzhang replied:
Removed

yaojiefeng commented on cdap-app-fabric/src/main/java/io/cdap/cdap/app/guice/AppFabricServiceRuntimeModule.java:
```diff
@@ -270,6 +273,7 @@ private AppFabricServiceModule(Class<? extends HttpHandler>... handlerClasses) {
     protected void configure() {
       bind(PipelineFactory.class).to(SynchronousPipelineFactory.class);
 
+      bind(PluginFinder.class).to(LocalPluginFinder.class);
```
why do we need the change here and remove the ones in `InMemoryProgramRunnerModule`?

wyzhang replied:
After the change, LocalApplicationManager used by AppFabric and PreviewApplicationManager used by Preview requires this binding. But Preview uses AppFabricServiceRuntime module, so have to move the binding to AppFabricServiceRuntimeModule to satisfy the need by both appfab and preview.

yaojiefeng commented on cdap-app-fabric/src/main/java/io/cdap/cdap/app/preview/DefaultPreviewRunnerModule.java:
```diff
@@ -46,6 +46,8 @@
 import io.cdap.cdap.internal.app.runtime.ProgramRuntimeProviderLoader;
 import io.cdap.cdap.internal.app.runtime.artifact.ArtifactRepository;
 import io.cdap.cdap.internal.app.runtime.artifact.ArtifactStore;
+import io.cdap.cdap.internal.app.runtime.artifact.LocalPluginFinder;
```
nit - unused

wyzhang replied:
Removed

yaojiefeng commented on cdap-app-fabric/src/main/java/io/cdap/cdap/app/guice/InMemoryProgramRunnerModule.java:
```diff
@@ -72,9 +72,7 @@ protected void configure() {
               .build(ArtifactManagerFactory.class));
     expose(ArtifactManagerFactory.class);
 
-    bind(PluginFinder.class).to(LocalPluginFinder.class);
```
nit - these imports can be removed

wyzhang replied:
Done

yaojiefeng reviewed (commented):
this `LocalArtifactLoaderStage` is still using `artifactRepository`, is there following pr to remove that?

wyzhang committed:
(CDAP-16353) Replace ArtifactRepository param with PluginFinder for AbstractConfigurer

Motivation:
AbstractConfigurer uses ArtifactRepository just to find the artifact for a given plugin,
this functionality is already provided by PluginFinder which has local and remote implementations,
thereby allowing us to use local impl when having shared SQL and remote version for non-shared
per service local levelDB.
